scripts/pre-commit/verify_migration_scripts.py

```diff
@@ -0,0 +1,182 @@
+    handler.setLevel(logging.INFO)
+    formatter = logging.Formatter("%(levelname)s - %(message)s")
+    handler.setFormatter(formatter)
+    logger.addHandler(handler)
+
+CODE_EXTENSION = ".sql"
+BOOTSTRAP_FILE = "bootstrap.sql"
+METADATA_STORE_SCRIPTS_DIR = "metadata_store_scripts"
+
+
+class MigrationValidationError(Exception):
+    """Raised when a migration validation check fails."""
+
+
+def is_code_file(filepath: str) -> bool:
+    return Path(filepath).suffix == CODE_EXTENSION
+
+
+def get_project_root() -> Path:
+    project_root = subprocess.run(
+        ["git", "rev-parse", "--show-toplevel"],
+        capture_output=True,
+        text=True,
+        check=True
+    )
+    return Path(project_root.stdout.strip())
+
+
+def parse_date(version_str: str) -> datetime:
+    date_str = ".".join(version_str[1:].split(".")[:3])
+    return datetime.strptime(date_str, "%Y.%m.%d")
+
+
+def check_duplicate_migration_versions(new_migration_files: tp.List, old_migration_files: tp.List) -> None:
+    """
+    Check if there are any duplicate migration versions
+    1. Between new and old migration files.
+    2. Between new migration files.
+    """
+    new_versions = [file.split("_")[0] for file in new_migration_files if file != BOOTSTRAP_FILE]
+    old_versions = [file.split("_")[0] for file in old_migration_files]
+
+    duplicates = set(new_versions) & set(old_versions)
+    if duplicates:
+        raise MigrationValidationError(
+            f"Duplicate migration versions found: {duplicates}"
+        )
+
+    new_migration_files_set = set(new_migration_files)
+    duplicates = [file for file in new_migration_files_set if new_versions.count(file.split("_")[0]) > 1]
+    if duplicates:
+        raise MigrationValidationError(
+            f"Duplicate migration versions found in new migration files: {duplicates}"
+        )
+
+
+def check_new_migration_file_format(new_migration_files: tp.List) -> None:
+    """
+    Check if the new migration files have a date format in their names.
+    """
+    for file in new_migration_files:
+        if file == BOOTSTRAP_FILE:
+            continue
+        if not re.match(r"^V\d{4}.\d{2}.\d{2}.\d{1,2}__.+\.sql$", file):
+            raise MigrationValidationError(
+                f"New migration file {file} does not have a valid file name format."
+            )
+
+
+def validate_new_migration_dates(new_migration_files: tp.List, old_migration_files: tp.List) -> None:
+    """
+    Validate that the new migration files have a date format in their names.
+    """
+    new_migration_date = [parse_date(file) for file in new_migration_files if file != BOOTSTRAP_FILE]
+    latest_migration_date = [parse_date(file) for file in old_migration_files]
+
+    invalid_new_migration_file_date = [
+        file_date for file_date in new_migration_date if file_date < max(latest_migration_date)
+    ]
+    if invalid_new_migration_file_date:
+        for file_date in invalid_new_migration_file_date:
```

Comment:
The for loop will break as soon as an exception is raised. Do we need to iterate over it.